### ISSUE TITLE
Add Lua Matricies?

### ISSUE THREAD

**Issue by awilliamson** [INPStarfall/Starfall#43]
Just wondering if adding matrices as a pure Lua module would be wanted. Would allow people to do some nice stuff without having to hack their own.

https://github.com/davidm/lua-matrix/blob/master

Was recommended from lua-users.org as a matrix library which handles real, complex and symbolics.

Checked the license, it allows for modification. Standard MIT. If we modify the standard matrix. Lua for use with libraries and include that license at the top as a comment that should suffice.

Thoughts and suggestions for this? ( We have quats for rotation why not matricies? )

My main reasoning behind this is that it can then be used to teach matricies to people who don't get the chance to be taught it or learn it in an interactive and visual way. Instant feedback from what they are doing will help them learn the power of matricies.

Of course we'll have to consider how this library would interact with Garry Vectors and its potential clash with VMatrix.

PS - Alternatively, we could write our own library in-house which is a more lightweight version tailored to GMod needs that we can build in support for other GLua data types.

**Comment by ColonelThirtyTwo** [comment]
I made the choice to use GLua's VMatrix type so that they could be used with the GLua API. Though last I checked, VMatrix was lacking a lot of functionality (I don't think there was a way to copy the matrix, or even retrieve individual cells).

As for including a Lua implementation, I don't see a need. Since SF runs Lua code, as long as the APIs match closely enough, a matrix implementation for regular Lua can run just as well when ran under SF. Thus, users can just drop a matrix.lua file in their scripts folder and include it like normal code.
